package/repos/forgejo.py

```diff
@@ -0,0 +1,209 @@
+from dataclasses import dataclass
+import logging
+from urllib.parse import urlparse
+
+import httpx
+
+from .base_handler import BaseHandler
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ForgejoMetadata:
+    archived: bool
+    archived_at: str | None
+    description: str
+    forks_count: int
+    stars_count: int
+    watchers_count: int
+
+
+@dataclass
+class ForgejoCommit:
+    sha: str
+    created: str
+    user: str | None
+
+
+class ForgejoClient:
+    """Client for interacting with Forgejo instances."""
+
+    def __init__(self, base_url: str):
+        self.base_url = base_url.rstrip("/")
+        self.api_base_url = f"{self.base_url}/api/v1"
+
+    def _build_url(self, path: str) -> str:
+        return f"{self.api_base_url}{path}"
+
+    def fetch_repository(self, repository: str) -> ForgejoMetadata | None:
+        url = self._build_url(f"/repos/{repository}")
+        try:
+            response = httpx.get(url)
+        except Exception as exp:
+            logger.error(exp)
+            return None
+
+        if response.status_code != httpx.codes.OK:
+            logger.error("Response code: %s for URL %s", response.status_code, url)
+            return None
+
+        data = response.json()
```

Comment:
A more concise and idiomatic approach here would be to use `raise_for_status`:

```suggestion
        try:
            response = httpx.get(url)
            response.raise_for_status()
            data = response.json()

        except (httpx.HTTPError, ValueError) as exc:
            logger.error("Failed to fetch %s: %s", url, exc)
            return None
```

This should catch all the httpx transport and general request errors via `httpx.HTTPError` while allowing more general errors to bubble up by now swallowing via a generic `Exception` catch.